engine/tests/ElggEntityTest.php

```diff
@@ -355,4 +355,23 @@ public function testCanAnnotateDoesntCallSpecificThenGenericHookForEmptyString()
 
 		$object->delete();
 	}
+
+	public function testCreateWithContainerGuidEqualsZero() {
+		$user = new \ElggUser();
+		$user->save();
+
+		$object = new \ElggObject();
+		$object->owner_guid = $user->guid;
+		$object->container_guid = 0;
+
+		// If container_guid attribute is not updated with owner_guid attribute
+		// ElggEntity::getContainerEntity() would return false
+		// thus terminating save()
+		$this->assertTrue($object->save());
+
+		$this->assertEqual($user->guid, $object->getContainerGUID());
+
+		$user->delete();
+
```

Comment:
Deleting a user deletes the object
On Dec 8, 2015 1:26 PM, "Steve Clay" <notifications@github.com> wrote:

> In engine/tests/ElggEntityTest.php
> <https://github.com/Elgg/Elgg/pull/9211#discussion_r46946688>:
>
> > +		$user = new \ElggUser();
> > +		$user->save();
> > +
> > +		$object = new \ElggObject();
> > +		$object->owner_guid = $user->guid;
> > +		$object->container_guid = 0;
> > +
> > +		// If container_guid attribute is not updated with owner_guid attribute
> > +		// ElggEntity::getContainerEntity() would return false
> > +		// thus terminating save()
> > +		$this->assertTrue($object->save());
> > +
> > +		$this->assertEqual($user->guid, $object->getContainerGUID());
> > +
> > +		$user->delete();
> > +
>
> Also delete $object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9211/files#r46946688>.
>